### DISCUSSION TITLE
Hi, I was trying to create a theme config, but causing ts issues,

### DISCUSSION THREAD

**Discussion by sayshark75** [chakra-ui/chakra-ui#8983] [Q&A] [answered]
- I created a latest nextjs app with ts and without tailwind,
- I added Chakra ui in it, as per v3 docs for nextjs app router, followed everything shown
- created a a theme.ts in themes path ```src/themes/theme.ts```
- added cli globally using ```npm i @chakra-ui/cli -g```
- added script to run the typegen in package.json ```"gen": "chakra typegen src/themes/theme.ts"```
- run the script ```yarn gen```
- generated the types 
```
┌  Chakra CLI ⚡️
│
◇  ✅ Generated conditions typings
│
◇  ✅ Generated recipe typings
│
◇  ✅ Generated utility typings
│
◇  ✅ Generated token typings
│
◇  ✅ Generated system types
│
└  🎉 Done!
```
- still theme.ts has typescript issues
- here is the file code
```
// ./src/themes/theme.ts

import { createSystem, defineConfig } from "@chakra-ui/react";

const config = defineConfig({
  theme: {
    colors: {},
  },
});

export default createSystem(config);
```
- the colors property has the ts issue as shown below
```
Object literal may only specify known properties, and 'colors' does not exist in type 'ThemingConfig'.ts(2353)
types.d.ts(230, 5): The expected type comes from property 'theme' which is declared here on type 'SystemConfig'
```
please let me know the issue, i am ready to provide more details...

**Comment by sayshark75** [comment]
- colors is inside tokens object, but not mentioned in docs ```theming > overview``` [here](https://www.chakra-ui.com/docs/theming/overview)
- should be like this? 
```
import { createSystem, defineConfig } from "@chakra-ui/react";

const config = defineConfig({
  theme: {
    tokens: {
      colors: {
        brand: {
          100: { value: "#e6f2ff" },
        },
      },
    }
  },
});

export default createSystem(config);
```

**Reply by segunadebayo** [reply, marked as answer]
Correct. It should be like that.

I've updated the docs directly.

**Reply by sayshark75** [reply]
I would like to update the docs but the write operation on the doc repo is stopped.

**Reply by y-wertz-priomold** [reply]
Adding to that topic I've found a similar issue modifying defaultVariants.

Example:
```typescript
import { createSystem, defaultConfig, defineConfig } from "@chakra-ui/react";

const config = defineConfig({
  theme: {
    recipes: {
      heading: {
        defaultVariants: {
          size: "5xl",
        },
      }
    }
  }
})
```

Which gives me the following typescript error:
![image](https://github.com/user-attachments/assets/b0082a5b-0354-483f-977a-f8a874a18d8c)
```
Object literal may only specify known properties, and 'size' does not exist in type '{ colorPalette?: ColorPalette | undefined; }'.ts(2353)
recipe.types.d.ts(47, 5): The expected type comes from property 'defaultVariants' which is declared here on type 'RecipeDefinition<RecipeVariantRecord>'
```

No matter which component it always say the only known property is `colorPalette`.

To make clear: It works perfectly fine and changes the default behaviour of Heading component, stil giving that typescript error, tho.
